common/utilities/tests/test_timing.py

```diff
@@ -0,0 +1,207 @@
+            time_mock.return_value = 5
+            res = await self.take_parameters_async("whew", 1, [2], {3: 3})
+            log_mock.assert_called_with('Method name=take_parameters_async took duration=5')
+            self.assertEqual("whew1", res)
+
+    @patch('common.utilities.timing.Stopwatch.stop_timer')
+    @patch('logging.info')
+    @async_test
+    async def test_async_times_execution_correctly(self, log_mock, time_mock):
+        time_mock.return_value = 0
+        task = self.default_method_async()
+
+        # check the method doesn't get timed until awaited
+        time_mock.return_value = 2
+
+        await task
+
+        log_mock.assert_called_with('Method name=default_method_async took duration=2')
+
+    @patch('common.utilities.timing.Stopwatch.stop_timer')
+    @patch('logging.info')
+    @async_test
+    async def test_invoke_with_time_varargs(self, log_mock, time_mock):
+        with self.subTest("Sync"):
+            time_mock.return_value = 5
+            res = self.var_parameters("whew", 1, 2, 3, 4, 5)
+            log_mock.assert_called_with('Method name=var_parameters took duration=5')
+            self.assertEqual("whew12345", res)
+        with self.subTest("Async"):
+            time_mock.return_value = 5
+            res = await self.var_parameters_async("whew", 1, "three", 4)
+            log_mock.assert_called_with('Method name=var_parameters_async took duration=5')
+            self.assertEqual("whew1three4", res)
+
+    @timing.time_function
+    def default_method(self):
+        return "slept"
+
+    @timing.time_function
+    async def default_method_async(self):
+        return 5
+
+    @timing.time_function
+    def throw_error_method(self):
+        raise ValueError("Whew")
+
+    @timing.time_function
+    async def throw_error_method_async(self):
+        raise ValueError("Whew")
+
+    @timing.time_function
+    def take_parameters(self, check, one, two, three):
+        assert check is not None
+        assert one is not None
+        assert two is not None
+        assert three is not None
+        return check + str(one)
+
+    @timing.time_function
+    async def take_parameters_async(self, check, one, two, three):
+        assert check is not None
+        assert one is not None
+        assert two is not None
+        assert three is not None
+        return check + str(one)
+
+    @timing.time_function
+    def var_parameters(self, *arg):
+        return ''.join([str(string) for string in arg])
+
+    @timing.time_function
+    async def var_parameters_async(self, *arg):
+        return ''.join([str(string) for string in arg])
+
+
+class FakeRequestHandler(RequestHandler):
+
+    @timing.time_request
+    def post(self):
+        self.write("hello")
+
+    @timing.time_request
+    async def get(self):
+        self.write("hello")
+
+    @timing.time_request
+    def put(self):
+        try:
+            raise ValueError("Whew")
+        finally:
+            self.write("put")
```

Comment:
What's the point of this line?

Reply:
I was trying to make it so all the methods I was calling return an actual value so that we can verify end result of the wrapped method is completed/returned, it was mentioned in a previous comment from Gareth that it would indicate that the async methods were correctly being called